docarray/utility/find.py

```diff
@@ -16,7 +17,7 @@ class FindResult(NamedTuple):
 
 
 def find(
-    index: DocumentArray,
+    index: Union[DocumentArray, DocumentArrayStacked],
```

Comment:
This seems a bit smelly to me. On the one hand, the user doesn't know that stacked and non stacked are two different classes, they just call stack() and unstack() on on object. On the other hand, they need to know about this distinction in cases like this

Reply:
I see your point probably we need to revisit the different concept that we expose to the user. Maybe what we need is a to expose the Abstract class under the name DocumentContainer

Reply:
Something like this, yes. Or go all the way and own `DocumentArray` and `DocumentList` with `stack()/unstack()` to go between them, but also telling this to the user. The way it currently is I don't think we should merge the PR, since it makes things more confusing on the outside (to our users) with the only benefit being on the inside (a refactoring)

Reply:
Actually I think we don't have a choice but to expose the Union. Because you need to call stack anyway inside your method if this is not yet stacked. And the stack method would only be available at the `DocuentArray` level so the Union will always be needed

Reply:
okay I have an idea let me try